iac-extensions/jvm-framework-config/src/main/java/org/sonar/iac/jvmframeworkconfig/checks/micronaut/HardcodedSecretsCheck.java

```diff
@@ -0,0 +1,210 @@
+    "micronaut.http.services.[^.]++.ssl.key-store.password",
+    "micronaut.http.services.[^.]++.ssl.trust-store.password",
+    "micronaut.chatbots.telegram.bots.[^.]++.token",
+    "flyway.datasources.[^.]++.password",
+    "nats.[^.]++.password",
+    "nats.[^.]++.token",
+    "nats.[^.]++.tls.trust-store-password",
+    "r2dbc.datasources.[^.]++.password",
+    "rabbitmq.servers.[^.]++.password",
+    "redis.servers.[^.]++.password",
+    "micronaut.security.token.jwt.encryptions.secret.[^.]++.secret",
+    "micronaut.security.token.jwt.signatures.secret.[^.]++.secret",
+    "micronaut.security.ldap.[^.]++.context.manager-password",
+    "micronaut.security.oauth2.clients.[^.]++.client-secret",
+    "datasources.[^.]++.password");
+  private static final Pattern PATTERN_PASSWORD_IN_CONNECTION_STRING = Pattern.compile("(?:;|^)AccountKey=(?<password>[a-zA-Z0-9+/=]{60,})(?:;|$)");
+  private static final Pattern PATTERN_PASSWORD_IN_URL = Pattern.compile("(?<protocol>[a-zA-Z]++)://(?<username>[^:@]++):(?<password>[^@]++)@.++");
+  private static final Map<String, Pattern> SENSITIVE_KEYS_WITH_PATTERN_VALUE = Map.of(
+    "azure.credential.storage-shared-key.connection-string", PATTERN_PASSWORD_IN_CONNECTION_STRING,
+    "micronaut.jms.activemq.artemis.connection-string", PATTERN_PASSWORD_IN_URL,
+    "micronaut.jms.activemq.classic.connection-string", PATTERN_PASSWORD_IN_URL,
+    "mqtt.client.server-uri", PATTERN_PASSWORD_IN_URL,
+    "rabbitmq.uri", PATTERN_PASSWORD_IN_URL);
+  // Extra use case: combine both wildcard in the key and URL with password in value
+  private static final Pattern PATTERN_SENSITIVE_RABBITMQ_PROPERTY = Pattern.compile("rabbitmq.servers.[^.]++.uri");
+
+  private static final Pattern VARIABLE = Pattern.compile("\\$\\{[^}]+}");
+  // Compiling patterns together is ~35% faster than checking patterns individually
+  private static final Predicate<String> PREDICATE_SENSITIVE_KEY_FULL_REGEX = Pattern.compile(String.join("|", SENSITIVE_KEYS_REGEX)).asMatchPredicate();
+
+  @Override
+  protected Set<String> sensitiveKeys() {
+    return SENSITIVE_KEYS;
+  }
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(Tuple.class, this::checkTuplePatternInKey);
+    init.register(Tuple.class, HardcodedSecretsCheck::checkTuplePatternInValue);
+    super.initialize(init);
+  }
+
+  private void checkTuplePatternInKey(CheckContext ctx, Tuple tuple) {
+    var key = tuple.key().value().value();
+    if (PREDICATE_SENSITIVE_KEY_FULL_REGEX.test(key)) {
+      var valueString = getStringValue(tuple);
+      if (valueString != null) {
+        checkValue(ctx, tuple, valueString);
+      }
+    } else {
+      var matcher = PATTERN_SENSITIVE_RABBITMQ_PROPERTY.matcher(key);
+      if (matcher.matches()) {
+        checkValueWithPattern(ctx, PATTERN_PASSWORD_IN_URL, tuple);
+      }
+    }
+  }
+
+  private static void checkTuplePatternInValue(CheckContext ctx, Tuple tuple) {
```

Comment:
Seems like it could be integrated into `checkTuplePatternInKey` / `checkTuple`

Reply:
I am not sure how to do that; although the logics are close, they are different, and I don't see how to achieve it.
Did you have a specific idea about it?

Reply:
Isn't it just another if else branch you could add to the mentioned method?

Reply:
Yes, but as it's the same registration of tuple.class, I think it fits there.

Reply:
Ok got it 👍